File: 프로그래머스/0/120835. 진료 순서 정하기/진료 순서 정하기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> emergency) {
    vector<int> answer;
    vector<int> sv = emergency;
    sort(sv.begin(), sv.end(), greater<int>());
    
    for (int i = 0; i < emergency.size(); i++) {
        for (int j = 0; j < sv.size(); j++) {
            if (emergency[i] == sv[j]) {
                answer.push_back(j + 1);
                break;
            }
        }
    }
    return answer;
}

File: 프로그래머스/0/120835. 진료 순서 정하기/진료 순서 정하기.cs
using System;

public class Solution {
    public int[] solution(int[] emergency) {
        int[] answer = new int[emergency.Length];
        int[] sorted = new int[emergency.Length];
        Array.Copy(emergency, sorted, emergency.Length);

        Array.Sort(sorted);
        Array.Reverse(sorted);

        for (int i = 0; i < emergency.Length; i++)
        {
            answer[i] = Array.IndexOf(sorted, emergency[i]) + 1;
        }
        
        return answer;
    }
}